lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -290,7 +286,7 @@ def parse(self, inventory, loader, path, cache=True):
         super(InventoryModule, self).parse(inventory, loader, path)
 
         config_data = {}
-        if self._validate_file(path):
+        if self.verify_file(path):
```

Comment:
Thanks for the work you've put in this. Looks great. :-)

Not a big deal, but this is a redundant call. You can just do `config_data = self._read_config_data(path)`. verify_file() is called when InventoryModule is instantiated by InventoryManager here: https://github.com/ansible/ansible/blob/1cf07028d400cd3cef5699c37cadfefb7bbb9532/lib/ansible/inventory/manager.py#L262

I added `import traceback; traceback.print_stack()` to aws_ec2 inventory plugin to verify_file to show the code path that calls that method:
```
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/bin/ansible-inventory", line 118, in <module>
    exit_code = cli.run()
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/lib/ansible/cli/inventory.py", line 136, in run
    self.loader, self.inventory, self.vm = self._play_prereqs(self.options)
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/lib/ansible/cli/__init__.py", line 804, in _play_prereqs
    inventory = InventoryManager(loader=loader, sources=options.inventory)
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/lib/ansible/inventory/manager.py", line 145, in __init__
    self.parse_sources(cache=True)
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/lib/ansible/inventory/manager.py", line 206, in parse_sources
    parse = self.parse_source(source, cache=cache)
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/lib/ansible/inventory/manager.py", line 262, in parse_source
    plugin_wants = bool(plugin.verify_file(source))
  File "/Users/shertel/Workspace/while_running_tests_3/ansible/lib/ansible/plugins/inventory/aws_ec2.py", line 464, in verify_file
    import traceback; traceback.print_stack()
```
And I can see from the fact that only one stack was printed that the method is just being called once.

You can test this too by adding `import traceback; traceback.print_stack()` to the beginning of your verify_file method. Right now you'll see that two stacks are printed out instead of one with two different code paths leading to the same method.